src/commons.c

```diff
@@ -196,13 +196,13 @@ str2enum (const GEnum map[], int len, const char *str) {
  *
  * On error, -1 is returned.
  * On success, the enumerated module value is returned. */
-char *
+const char *
 enum2str (const GEnum map[], int len, int idx) {
   int i;
 
   for (i = 0; i < len; ++i) {
     if (idx == map[i].idx)
-      return xstrdup (map[i].str);
+      return map[i].str;
```

Comment:
I'm getting an invalid pointer without the strdup. Are you seeing this on your end too?  Thanks!

```
./goaccess ../10.log -o a.html --log-format=VCOMBINED
free(): invalid pointer
Aborted (core dumped)
Thread 1 "goaccess" received signal SIGABRT, Aborted.
```

Reply:
No, I am not seeing such memory corruption.
Do you use some non default settings in the global goaccess.conf?
Can you share the stacktrace of the crash or a public version of the triggering input log?
Otherwise the following patch might resolve it?
```diff
---
 src/gkhash.c | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/src/gkhash.c b/src/gkhash.c
index d40e942d..3109c78c 100644
--- a/src/gkhash.c
+++ b/src/gkhash.c
@@ -1040,8 +1040,10 @@ inc_su64 (khash_t (su64) *hash, const char *key, uint64_t inc) {
     dupkey = xstrdup (key);
     k = kh_put (su64, hash, dupkey, &ret);
     /* operation failed */
-    if (ret == -1)
+    if (ret == -1) {
+      free(dupkey);
       return -1;
+    }
   } else {
     value = kh_val (hash, k) + inc;
   }
@@ -1087,6 +1089,7 @@ static uint32_t
 inc_si32 (khash_t (si32) *hash, const char *key, uint32_t inc) {
   khint_t k;
   int ret;
+  char *dupkey = NULL;
 
   if (!hash)
     return 0;
@@ -1094,10 +1097,16 @@ inc_si32 (khash_t (si32) *hash, const char *key, uint32_t inc) {
   k = kh_get (si32, hash, key);
   /* key not found, put a new hash with val=0 */
   if (k == kh_end (hash)) {
-    k = kh_put (si32, hash, key, &ret);
+    dupkey = xstrdup(key);
+    k = kh_put (si32, hash, dupkey, &ret);
     /* operation failed */
-    if (ret == -1)
+    if (ret == -1) {
+      free(dupkey);
       return 0;
+    }
+    /* concurrently added */
+    if (ret == 0)
+      free(dupkey);
     kh_val (hash, k) = 0;
   }
 
@@ -1484,9 +1493,7 @@ ht_inc_cnt_overall (const char *key, uint32_t val) {
   if (!hash)
     return 0;
 
-  if (get_si32 (hash, key) != 0)
-    return inc_si32 (hash, key, val);
-  return inc_si32 (hash, xstrdup (key), val);
+  return inc_si32 (hash, key, val);
 }
 
 int
@@ -1509,9 +1516,7 @@ ht_ins_seq (khash_t (si32) *hash, const char *key) {
   if (!hash)
     return 0;
 
-  if (get_si32 (hash, key) != 0)
-    return inc_si32 (hash, key, 1);
-  return inc_si32 (hash, xstrdup (key), 1);
+  return inc_si32 (hash, key, 1);
 }
 
 /* Insert an IP hostname mapped to the corresponding hostname.
-- 
2.43.0
```

Reply:
The patch didn't resolve the issue. However, to reproduce it, you'll need to build with `--enable-debug`. Without this flag, the issue doesn't occur. I'll incorporate some additional tests into the build workflow to help catch future issues. e.g., 

    ./configure --enable-utf8 --enable-geoip=mmdb --enable-debug --with-getline--with-openssl
    echo 'google.com.com:80 192.168.88.1 - - [27/Apr/2024:16:29:29 -0500] "GET / HTTP/1.1" 403 1431 "-" "Mozilla/5.0 (X11 ; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36" 1803' | ./goaccess - --log-format=VCOMBINED